src/javascript/app_2/Modules/Trading/Components/Form/TradeParams/Duration/simple_duration.jsx

```diff
@@ -2,18 +2,56 @@ import { PropTypes as MobxPropTypes } from 'mobx-react';
 import PropTypes                      from 'prop-types';
 import React, { Fragment }            from 'react';
 import ButtonToggleMenu               from 'App/Components/Form/button_toggle_menu.jsx';
+import DatePicker                     from 'App/Components/Form/DatePicker';
 import InputField                     from 'App/Components/Form/input_field.jsx';
 import RangeSlider                    from 'App/Components/Form/RangeSlider';
+import { hasIntradayDurationUnit }    from 'Stores/Modules/Trading/Helpers/duration';
+import {
+    isTimeValid,
+    setTime,
+    toMoment }                        from 'Utils/Date';
 
 const SimpleDuration = ({
     changeDurationUnit,
+    duration_d,
+    duration_t,
+    duration_units_list,
+    duration_min_max,
+    expiry_date,
     getDurationFromUnit,
     number_input_props,
-    duration_units_list,
     shared_input_props,
     simple_duration_unit,
-    duration_t,
+    server_time,
+    start_date,
+    start_time,
 }) => {
+    /* get max and min date for datepicker */
+    const moment_expiry      = toMoment(expiry_date || server_time);
+    let is_24_hours_contract = false;
+    let max_date_duration,
+        min_date_expiry;
+    const max_daily_duration = duration_min_max.daily ? duration_min_max.daily.max : 365 * 24 * 3600;
+    const moment_contract_start_date_time =
+            setTime(toMoment(start_date || server_time), (isTimeValid(start_time) ? start_time : server_time.format('HH:mm')));
+    const has_intraday_duration_unit = hasIntradayDurationUnit(duration_units_list);
+
+    is_24_hours_contract = (!!start_date || moment_expiry.isSame(toMoment(server_time), 'day')) && has_intraday_duration_unit;
+
+    if (is_24_hours_contract) {
+        min_date_expiry = moment_contract_start_date_time.clone().startOf('day');
+        max_date_duration = moment_contract_start_date_time.clone().add(
+            start_date ? 24 * 3600 : (max_daily_duration), 'second');
+    } else {
+        min_date_expiry = moment_contract_start_date_time.clone().startOf('day');
+        max_date_duration = moment_contract_start_date_time.clone().add(max_daily_duration, 'second');
+
+        if (!has_intraday_duration_unit) {
+            min_date_expiry.add(1, 'day');
+        }
+    }
```

Comment:
Right now all this logic is duplicated in `advanced_duration` also, perhaps composing a new `<DaysInputField>` component with Datepicker and InputField could solve this?